### PR TITLE
feat(distortion_corrector): use gyroscope for correcting LiDAR distortion

### PR DESCRIPTION
**Pull request by kminoda**
Signed-off-by: kminoda <koji.m.minoda@gmail.com>

## Description
We want to use gyroscope data for distortion_corrector since the yaw rate calculated from vehicle steering may be noisy depending on the type of the vehicle.

In the proposed implementation, distortion_corrector will subscribe to two twist topics:
- /sensing/imu/imu_data
- /sensing/vehicle_velocity_converter/twist_with_covariance

174419885-ae4b9630-aed9-43c5-970e-71e63fad4174

When `use_imu` is set true and IMU data is available, the tool corrects the LiDAR distortion using a gyroscope (velocity is used from the odometry as usual).

Expected behavior is as follows:
```
if gyroscope is available && use_imu:
  v <- twist
  w <- gyroscope
else:
  v <- twist
  w <- twist
```

## Related links
- (We replaced https://github.com/autowarefoundation/autoware.universe/pull/1053 with this PR)
- https://github.com/autowarefoundation/autoware.universe/pull/1120
- https://github.com/tier4/aip_launcher/pull/67
- https://github.com/tier4/autoware_launch/pull/365
- https://github.com/autowarefoundation/sample_sensor_kit_launch/pull/36

## Tests performed
I used a rosbag in which distortion correction failed due to the large jump in `/vehicle/status/velocity_status/heading_rate`, and verified that the proposed algorithm addresses the issue.

Before: We can observe that the LiDAR pointcloud (in yellow) is a bit skewed
Screenshot from 2022-06-16 15-01-25

After: The skew is successfully mitigated
Screenshot from 2022-06-16 15-01-50

I have also confirmed that the output of gyro_odometer is as expected.
Screenshot from 2022-06-27 18-06-58

I also verified that by using a print debugging, `/vehicle/status/velocity_status/heading_rate` is successfully replaced with yaw rate from the gyroscope during the distortion correction step.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: sensing/pointcloud_preprocessor/src/distortion_corrector/distortion_corrector.cpp [hunk 1]
```diff
@@ -159,34 +199,58 @@ bool DistortionCorrectorComponent::undistortPointCloud(
...
-    float w{static_cast<float>(velocity_report_it->heading_rate)};
+    float v{static_cast<float>(twist_it->twist.linear.x)};
+    float w{static_cast<float>(twist_it->twist.angular.z)};
 
-    if (std::abs(*it_time_stamp - rclcpp::Time(velocity_report_it->header.stamp).seconds()) > 0.1) {
+    if (std::abs(*it_time_stamp - rclcpp::Time(twist_it->header.stamp).seconds()) > 0.1) {
       RCLCPP_WARN_STREAM_THROTTLE(
         get_logger(), *get_clock(), 10000 /* ms */,
-        "velocity_report time_stamp is too late. Cloud not interpolate.");
+        "twist time_stamp is too late. Cloud not interpolate.");
```
Comment (YamatoAndo) [inline sensing/pointcloud_preprocessor/src/distortion_corrector/distortion_corrector.cpp:234]:
```suggestion
        "twist time_stamp is too late. Could not interpolate.");
```

File: sensing/pointcloud_preprocessor/src/distortion_corrector/distortion_corrector.cpp [hunk 2]
```diff
@@ -83,7 +124,7 @@ void DistortionCorrectorComponent::onPointCloud(PointCloud2::UniquePtr points_ms
   tf2::Transform tf2_base_link_to_sensor{};
   getTransform(points_msg->header.frame_id, base_link_frame_, &tf2_base_link_to_sensor);
 
-  undistortPointCloud(velocity_report_queue_, tf2_base_link_to_sensor, *points_msg);
+  undistortPointCloud(tf2_base_link_to_sensor, *points_msg);
 
   if (points_sub_count > 0) {
```
Comment (YamatoAndo) [inline sensing/pointcloud_preprocessor/src/distortion_corrector/distortion_corrector.cpp:129]: At line 120, we confirm that `points_sub_count` is less than 1.
```
  if (points_sub_count < 1) {
    return;
  }
```
So, I think that this if-statement at 129 line is unnecessary.

**Comment by YamatoAndo** [comment]
Please fix the CI/CD checks.

File: sensing/pointcloud_preprocessor/src/distortion_corrector/distortion_corrector.cpp (hunk 1 above)
Comment (kminoda) [reply sensing/pointcloud_preprocessor/src/distortion_corrector/distortion_corrector.cpp]: https://github.com/autowarefoundation/autoware.universe/pull/1120/commits/510fb12f2b5df5cc8dd1c5588990887d177e8bf8
Reflected!

File: sensing/pointcloud_preprocessor/src/distortion_corrector/distortion_corrector.cpp (hunk 2 above)
Comment (kminoda) [reply sensing/pointcloud_preprocessor/src/distortion_corrector/distortion_corrector.cpp]: https://github.com/autowarefoundation/autoware.universe/pull/1120/commits/510fb12f2b5df5cc8dd1c5588990887d177e8bf8
Reflected!

**Review by YamatoAndo — APPROVED** [review]
The code is a bit complicated, but I think this is fine for now.

**Comment by kminoda** [comment]
Related discussion: https://github.com/orgs/autowarefoundation/discussions/2643

**Pull request merged**